ballerina-tests/tests/01_listener_test.bal

```diff
@@ -60,3 +61,100 @@ function testAttachingGraphQLServiceWithAnnotationToDynamicListener() returns er
+                age: 50
+            }
+        }
+    };
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["listener"]
+}
+function testAttachServiceWithMutationToHttp2BasedListener() returns error? {
+    string document = string `mutation { setName(name: "Heisenberg") { name } }`;
+    string url = "http://localhost:9190/service_with_http2";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = {
+        data: {
+            setName: {
+                name: "Heisenberg"
+            }
+        }
+    };
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["listener"]
+}
+function testAttachServiceWithSubscriptionToHttp2BasedListener() returns error? {
+    graphql:Error? result = http2BasedListener.attach(subscriptionService);
+    test:assertTrue(result is graphql:Error);
+    graphql:Error err = <graphql:Error>result;
+    string expecctedMessage = string `Websocket listener initialization failed due to the incompatibility of ` +
+                              string `provided HTTP(version 2.0) listener`;
+    test:assertEquals(err.message(), expecctedMessage);
+}
+
+@test:Config {
+    groups: ["listener"]
+}
+function testAttachServiceWithQueryToHttp1BasedListener() returns error? {
+    string document = string `query { person{ age } }`;
+    string url = "http://localhost:9191/service_with_http1";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = {
+        data: {
+            person: {
+                age: 50
+            }
+        }
+    };
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["listener"]
+}
+function testAttachServiceWithMutationToHttp1BasedListener() returns error? {
+    string document = string `mutation { setName(name: "Heisenberg") { name } }`;
+    string url = "http://localhost:9191/service_with_http1";
+    json actualPayload = check getJsonPayloadFromService(url, document);
+    json expectedPayload = {
+        data: {
+            setName: {
+                name: "Heisenberg"
+            }
+        }
+    };
+    assertJsonValuesWithOrder(actualPayload, expectedPayload);
+}
+
+@test:Config {
+    groups: ["listener"]
+}
+function testAttachServiceWithSubscriptionToHttp1BasedListener() returns error? {
+    string document = string `subscription { messages }`;
+    string url = "ws://localhost:9191/service_with_http1";
+    websocket:Client wsClient1 = check new(url);
+    websocket:Client wsClient2 = check new(url);
+    check writeWebSocketTextMessage(document, wsClient1);
+    check writeWebSocketTextMessage(document, wsClient2);
+    foreach int i in 1 ..< 4 {
+        json expectedPayload = {data: {messages: i}};
+        check validateWebSocketResponse(wsClient1, expectedPayload);
+        check validateWebSocketResponse(wsClient2, expectedPayload);
+    }
+}
```

Comment:
We should check the Query and Mutation with HTTP_1 client.

Reply:
Already added tests for Query and Mutations.

Reply:
Those tests are using the HTTP 1.1 client right?

Reply:
Missed that. Added in https://github.com/ballerina-platform/module-ballerina-graphql/pull/1034/commits/9f06b729e18b9a235224a4073889e2ed3f60d58e